streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java

```diff
@@ -75,8 +76,10 @@
      * Registers and possibly restores the specified storage engine.
      *
      * @param store the storage engine
+     * @throws IllegalStateException If store get's registered after initialized is already finished
+     * @throws StreamsException if the store's change log does not contain the partition
      */
-    void register(StateStore store, boolean loggingEnabled, StateRestoreCallback stateRestoreCallback);
+    void register(StateStore store, StateRestoreCallback stateRestoreCallback);
```

Comment:
We need to mark it in the upgrade guide docs as it is public API and this change is not backward compatible.

Reply:
This is basically a consequence of KIP-182 -- I think, we should update the KIP and send a note to the dev list. \cc @dguy (can you take care of this?) Having said this, we need to keep this parameter for now to not break the API and mark this parameter as "deprecated/ignored" in the JavaDocs. Or can we still use it, and enable logging on the store if it is set?

Reply:
I'm not sure how this relates to KIP-182. That variable has been unused for quite some time.

Reply:
I see. So no regression. That's good. I thought that the changes with regard to new `StoreBuilder` made this one obsolete.

I'll update the JavaDocs accordingly and revert the change. Not sure, how/if/when we can change the interface to get rid of this one -- it's a breaking change. Created https://issues.apache.org/jira/browse/KAFKA-5974